### PR TITLE
[Ready for Review]: IAE-16876 Fix 508 Accessibility and JAWS issues with actions-dropwdown component

### PR DESCRIPTION
**Pull request by anuragmundada-reisys**
## Description
Fixed 508 and JAWS issues with sam-actions-dropown component following the guidelines from https://www.w3.org/TR/wai-aria-practices-1.1/examples/menu-button/menu-button-actions.html

## Motivation and Context
https://cm-jira.usa.gov/browse/IAE-16876

## Type of Change (Select One and Apply Github Label)

- [x] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [ ] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Which browsers have you tested?
- [x] Internet Explorer 11
- [x] Edge
- [x] Chrome
- [x] Firefox
- [ ] Safari

## Checklist:

- [x] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [ ] My code passes the automated linter.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] My code is 508 compliant as tested by AMP and JAWS
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts [hunk 1]
```diff
@@ -48,17 +48,24 @@ export class SamActionsDropdownComponent {
...
     return this.showActions = false;
   }
 
   toggleActions() {
-    if(!this.showActions){
+    this.showActions = !this.showActions;
+    if(this.showActions) {
+      this.setFocusOnFirstItem();
+    } else {
+      console.log("here reset focusIndex")
```
Comment (yerramshilpa) [inline src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts]: @anuragmundada-reisys Can you remove the console log

File: src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts [hunk 2]
```diff
@@ -75,26 +82,50 @@ export class SamActionsDropdownComponent {
...
       if(this.focusIndex-1>=0){
         this.focusIndex--;
         this.actionsList.toArray()[this.focusIndex].nativeElement.focus();
+      } else {
+        this.focusIndex = this.actionsList.toArray().length - 1;
+        this.actionsList.toArray()[this.focusIndex].nativeElement.focus();
       }
       event.preventDefault();
       event.stopPropagation();
+    } else if (KeyHelper.is("esc", event)){
```
Comment (yerramshilpa) [inline src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts]: Can you add type information for the function parameters and the return type?

File: src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts (hunk 1 above)
Comment (yerramshilpa) [reply src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts]: Add test cases for the new functionality.

Comment (yerramshilpa) [reply src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts]: Can you also make sure cross-browser testing? It works all browsers.

File: src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts (hunk 2 above)
Comment (anuragmundada-reisys) [reply src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts]: Which function parameters are you referring to? @yerramshilpa

File: src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts (hunk 1 above)
Comment (anuragmundada-reisys) [reply src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts]: Done

File: src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts (hunk 2 above)
Comment (yerramshilpa) [reply src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts]: I am refering to if (KeyHelper.is("esc", event)) condition check, can you changes some like below.

import { KeyHelper, KEYS } from '../../utilities/key-helper/key-helper';
if (KeyHelper.is(KEYS.SPACE, evt))

Comment (anuragmundada-reisys) [reply src/ui-kit/components/actions/actions-dropdown/actions-dropdown.component.ts]: Done

**Pull request merged**